src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef CLP_FFI_PY_IR_NATIVE_PYKEYVALUEPAIRLOGEVENT_HPP
+#define CLP_FFI_PY_IR_NATIVE_PYKEYVALUEPAIRLOGEVENT_HPP
+
+#include <clp_ffi_py/Python.hpp>  // Must always be included before any other header files
+
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <gsl/gsl>
+
+#include <clp_ffi_py/PyObjectUtils.hpp>
+
+namespace clp_ffi_py::ir::native {
+/**
+ * A PyObject structure functioning as a Python-compatible interface to retrieve a key-value pair
+ * log event. The underlying data is pointed to by `m_kv_pair_log_event`.
+ */
+class PyKeyValuePairLogEvent {
+public:
+    // Delete default constructor to disable direct instantiation.
+    PyKeyValuePairLogEvent() = delete;
+
+    // Delete copy/move constructors and assignment operators
```

Comment:
```suggestion
    // Delete copy & move constructors and assignment operators
```

Copied from our internal guidelines.